### DISCUSSION TITLE
Wifi doesn't work after installing Elementary OS on a Macbook Pro Early 2011

### DISCUSSION THREAD

**Discussion by sporlab** [elementary/triage#121] [Q&A]
Hi

I just installed Elementary OS on my old Macbook Pro Early 2011 for my kids to make them familiar with computers and Linux. It worked quite well with most of the hardware and performs much faster but my problem is that it is not recognizing wifi module of the laptop. It is not showing any option for wifi. 
The network controller is Broadcom Inc. BCM4331  802.11a/b/g/n (rev02)
Wi-Fi:
Type   IEEE80211
BSD Device Name:   en1
Configuration method:    DHCP

Is there any way to get connected to Wifi? It used to work just well on mac os.

Thanks

**Comment by xnacly** [comment]
https://askubuntu.com/questions/55868/installing-broadcom-wireless-drivers this should fix your issue

**Comment by mpanhans** [comment]
The solution is to (1) get internet connectivity somehow, whether it be an ethernet cable or bluetooth adapter, (2) install the bcmwl-kernel-source driver through AppCenter, and (3) then reboot the machine.

**Comment by chrisric** [comment]
I realize this is an old thread but might help someone in the future.

I installed elementary OS 8 on an old mac mini and the wifi was detected but could not join. It would just try joining over and over, after much troubleshooting I realized it was trying to use WPA3 which does not work with these old drivers.

I changed the GUI setting to "WPA/WPA2/WPA3 Personal" and it started working.

Below is my hardware for reference, might help someone else that is stuck.

```
Model Name:	Mac mini
  Model Identifier:	Macmini6,1
  Processor Name:	Dual-Core Intel Core i5
  Processor Speed:	2.5 GHz
  Number of Processors:	1
  Total Number of Cores:	2
  L2 Cache (per Core):	256 KB
  L3 Cache:	3 MB
  Hyper-Threading Technology:	Enabled
  Memory:	16 GB

$ sudo wpa_cli status
Selected interface 'wlp2s0'
bssid=<snip>
freq=5745
ssid=Skynet
id=0
mode=station
wifi_generation=4
pairwise_cipher=CCMP
group_cipher=CCMP
key_mgmt=WPA2-PSK
pmf=1
mgmt_group_cipher=BIP
wpa_state=COMPLETED
ip_address=<snip>
address=<snip>
uuid=<snip>

$ ethtool -i wlp2s0
driver: wl0
version: 6.30.223.271 (r587334)
firmware-version:
expansion-rom-version:
bus-info: 0000:02:00.0
supports-statistics: no
supports-test: no
supports-eeprom-access: no
supports-register-dump: no
supports-priv-flags: no

$ lspci -vnn
02:00.0 Network controller [0280]: Broadcom Inc. and subsidiaries BCM4331 802.11a/b/g/n [14e4:4331] (rev 02)
	Subsystem: Broadcom Inc. and subsidiaries BCM4331 802.11a/b/g/n [14e4:4331]
	Flags: bus master, fast devsel, latency 0, IRQ 17, IOMMU group 13
	Memory at a0600000 (64-bit, non-prefetchable) [size=16K]
	Capabilities: <access denied>
	Kernel driver in use: wl
	Kernel modules: bcma, wl
```
